src/drivers/intel/ssp/ssp.c

```diff
@@ -726,11 +733,18 @@ static void ssp_post_stop(struct dai *dai)
 	/* release clocks if SSP is inactive */
 	if (ssp->state[SOF_IPC_STREAM_PLAYBACK] != COMP_STATE_ACTIVE &&
 	    ssp->state[SOF_IPC_STREAM_CAPTURE] != COMP_STATE_ACTIVE) {
-		dai_info(dai, "releasing BCLK/MCLK clocks for SSP%d...", dai->index);
+		if (!(ssp->params.clks_control &
+			 SOF_DAI_INTEL_SSP_CLKCTRL_BCLK_ES)) {
+			dai_info(dai, "ssp_post_stop releasing BCLK clocks for SSP%d...", dai->index);
 #if CONFIG_INTEL_MN
-		mn_release_bclk(dai->index);
+			mn_release_bclk(dai->index);
 #endif
-		mn_release_mclk(ssp->config.ssp.mclk_id);
+		}
+		if (!(ssp->params.clks_control &
+			 SOF_DAI_INTEL_SSP_CLKCTRL_MCLK_ES)) {
+			dai_info(dai, "ssp_post_stop releasing MCLK clocks for SSP%d...", dai->index);
+			mn_release_mclk(ssp->config.ssp.mclk_id);
```

Comment:
We should have some sort of symmetry in the function names between bclk_set and release_bclk? get/put, acquire/release, etc.

Reply:
> We should have some sort of symmetry in the function names between bclk_set and release_bclk? get/put, acquire/release, etc.

added ssp_release_bclk and ssp_release_mclk